_pytest/python_api.py

```diff
@@ -6,10 +6,271 @@
 from _pytest.compat import isclass
 from _pytest.runner import fail
 import _pytest._code
+
 # builtin pytest.approx helper
 
+class ApproxBase(object):
+    """
+    Provide shared utilities for making approximate comparisons between numbers 
+    or sequences of numbers.
+    """
+
+    def __init__(self, expected, rel=None, abs=None):
+        self.expected = expected
+        self.abs = abs
+        self.rel = rel
+
+    def __repr__(self):
+        return ', '.join(
+                repr(self._approx_scalar(x))
+                for x in self._yield_expected())
+
+    def __eq__(self, actual):
+        return all(
+                a == self._approx_scalar(x)
+                for a, x in self._yield_comparisons(actual))
+
+    __hash__ = None
+
+    def __ne__(self, actual):
+        return not (actual == self)
+
+    def _approx_scalar(self, x):
+        return ApproxScalar(x, rel=self.rel, abs=self.abs)
+
+    def _yield_expected(self, actual):
+        """
+        Yield all the expected values associated with this object.  This is 
+        used to implement the `__repr__` method.
+        """
+        raise NotImplementedError
+
+    def _yield_comparisons(self, actual):
+        """
+        Yield all the pairs of numbers to be compared.  This is used to 
+        implement the `__eq__` method.
+        """
+        raise NotImplementedError
+
+
+
+try:
+    import numpy as np
+
+    class ApproxNumpy(ApproxBase, np.ndarray):
+        """
+        Perform approximate comparisons for numpy arrays.
+
+        This class must inherit from numpy.ndarray in order to allow the approx 
```

Comment:
I was thinking about this again... the problem on forcing it to be on the left-side is that it would be weird to use the standard `assert obtained == expected`, since the "approximation" is usually done on the "expected" value, i.e., this makes much more sense and is more easier to read:

```python
assert calculate_something() == approx(5.0, rtol=1e-3)  # must be in range 5.0 +- 0.005
```